### ISSUE TITLE
Numeric#round(n, mode) - round modes are not supported

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1509]
At least Ruby 2.4 supports several round modes. See https://ruby-doc.org/stdlib-2.4.0/libdoc/bigdecimal/rdoc/BigDecimal.html#method-c-mode

Unfortunately, TruffleRuby triggers `TypeError: Truffle doesn't have a case for the org.truffleruby.stdlib.bigdecimal.BigDecimalNodesFactory$RoundNodeFactory$RoundNodeGen node with values of type` exception with any used round mode.

To reproduce:

```ruby
require 'bigdecimal'

BigDecimal('1.516').round(2, :half_even)
```

Error:

```
TypeError: Truffle doesn't have a case for the org.truffleruby.stdlib.bigdecimal.BigDecimalNodesFactory$RoundNodeFactory$RoundNodeGen node with values of type  Truffle::BigDecimal(com.oracle.truffle.object.basic.DynamicObjectBasic) java.lang.Integer=2 Symbol(com.oracle.truffle.object.basic.DynamicObjectBasic)
	from (irb):5:in `round'
	from (irb):5
	from ~/.rubies/truffleruby-1.0.0-rc10/bin/irb:29:in `<main>'
```

**Comment by nirvdrum** [comment]
Thanks for the report. TruffleRuby does work with some round modes. It passes all the [BigDecimal#round specs](https://github.com/oracle/truffleruby/blob/master/spec/ruby/library/bigdecimal/round_spec.rb). It looks like the symbol-based modes are problematic. We probably need a lookup table to translate them to integer values. It shouldn't be too difficult.

It looks like we should also improve the [Ruby Spec Suite](https://github.com/ruby/spec) by adding more cases to test.

**Comment by nirvdrum** [comment]
Are you sure that's supposed to work though? In MRI 2.4.x and 2.5.x, I get an `ArgumentError`:
```
> BigDecimal('1.516').round(2, :round_half_even)
ArgumentError: invalid rounding mode
	from (irb):2:in `round'
	from (irb):2
	from /home/kjmenard/.rbenv/versions/2.4.4/bin/irb:11:in `<main>'
```

So it looks like we might just need raise an `ArgumentError` with a better message.

**Comment by deepj** [comment]
@nirvdrum You have a typo. It's `half_even`, not `round_half_even` :)

**Comment by deepj** [comment]
![screenshot 2018-12-07 at 23 43 21](https://user-images.githubusercontent.com/33736/49676424-91d98400-fa71-11e8-9cf8-6ab790b5a435.png)

**Comment by nirvdrum** [comment]
Right you are. I messed up translating `BigDecimal::ROUND_HALF_EVEN` to a symbol. Thanks for pointing that out.

**Comment by deepj** [comment]
@eregon @chrisseaton Could this be a priority for RC14, please? In fact, this is one of issues why I started to report bugs in TruffleRuby. I'd like to run a quite trivial script importing exchange rates on TruffleRuby one day :)

**Comment by chrisseaton** [comment]
Sorry this hasn't managed to go into RC15, but we have a lot of time scheduled this month to look at this issue backlog.

**Comment by chrisseaton** [comment]
Working on this now.

**Comment by chrisseaton** [comment]
Fix for this done and on the way.

**Comment by chrisseaton** [comment]
Fixed in 0beec8abe77f5aafac00f983fe67d77f8c573ac8.
